src/components/instant-apps-interactive-legend/support/helpers.ts

```diff
@@ -811,11 +813,10 @@ export function getTheme(el: HTMLElement): string {
 }
 
 // store
-export function updateStore(data: IInteractiveLegendData, layerData?: { intLegendLayerData: IIntLegendLayerData; layerId: string }): void {
-  const layerId = layerData?.layerId;
+export function updateStore(layerData: { intLegendLayerData: IIntLegendLayerData; layerId: string }): void {
+  const layerId = layerData?.layerId as string;
```

Comment:
You could wrap all the lines in this function in an if statement `if(layerData != null)`, then you wouldn't need the optional chaining. If `layerId` is undefined or null you wouldn't want to run `store.set(…)`

Reply:
Good point! Done